### PR TITLE
fix: atom crash on Glib-GIO-ERROR

### PR DESCRIPTION
**Pull request by ysndr**
###### Motivation for this change
in  the recent unstable releases of nixos atom used to crash when opening file open/save dialogs.
This might be related to a similar issue happening with sakura (#21698, #21700) which is also the inspiration for this fix

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/applications/editors/atom/default.nix [hunk 1]
```diff
@@ -24,7 +24,9 @@ let
       sed -i "s/${pname})/.${pname}-wrapped)/" $out/bin/${pname}
       # sed -i "s/'${pname}'/'.${pname}-wrapped'/" $out/bin/${pname}
       wrapProgram $out/bin/${pname} \
-        --prefix "PATH" : "${gvfs}/bin"
+        --prefix "PATH" : "${gvfs}/bin" \
+        --suffix "XDG_DATA_DIRS" : "${gtk3}/share/gsettings-schemas/${gtk3.name}/";
```
Comment (Mic92) [inline pkgs/applications/editors/atom/default.nix]: @jtojnar I guess wrapGAppsHook is more robust here?

Comment (jtojnar) [reply pkgs/applications/editors/atom/default.nix]: Yep, now that electron uses gtk3, `wrapGAppsHook` would work. Although, the PATH would need to be added to `gappsWrapperArgs`.

Comment (ysndr) [reply pkgs/applications/editors/atom/default.nix]: alright, will use  `wrapGAppsHook` then
about `gappsWrapperArgs`: what is supposed to be put in there? like `--prefix "PATH" : "${gvfs}/bin"?`

Comment (Mic92) [reply pkgs/applications/editors/atom/default.nix]: it takes the same arguments as `wrapProgram`.

**Pull request merged**